sechub-administration/src/main/java/com/mercedesbenz/sechub/domain/administration/user/AnonymousUserRestController.java

```
@@ -0,0 +1,85 @@
+    // todo: we should define a default redirect uri which is always available
+    // (/login is not)
+    private final String DEFAULT_REDIRECT_URI = "/api/anonymous/check/alive";
+    private final String FRONTEND_REDIRECT_URI_PATH = "/user/email_verify";
+
+    AnonymousUserRestController(UserEmailAddressUpdateService emailAddressUpdateService, SecHubSecurityProperties secHubSecurityProperties) {
+        this.emailAddressUpdateService = emailAddressUpdateService;
+        this.secHubSecurityProperties = secHubSecurityProperties;
+        REDIRECT_URI = getRedirectUri(secHubSecurityProperties);
+    }
+
+    /* @formatter:off*/
+    @UseCaseAnonymousUserVerifiesEmailAddress(
+            @Step(
+                    number = 1,
+                    name = "Rest call",
+                    next = {2},
+                    description = "User verifies his new email address",
+                    needsRestDoc = true))
+    @RequestMapping(value = AdministrationAPIConstants.API_ANONYMOUS_USER_VERIFY_EMAIL + "/{oneTimeToken}", method = RequestMethod.GET)
+    public void verifyEmailAddress(@PathVariable(name = "oneTimeToken") String oneTimeToken, HttpServletResponse response) throws IOException {
+        /* @formatter:on */
+        emailAddressUpdateService.userVerifiesUserEmailAddress(oneTimeToken);
+        response.sendRedirect(REDIRECT_URI);
+    }
+
+    private String getRedirectUri(SecHubSecurityProperties secHubSecurityProperties) {
+        SecHubSecurityProperties.LoginProperties loginProperties = secHubSecurityProperties.getLoginProperties();
+        if (loginProperties == null) {
+            return DEFAULT_REDIRECT_URI;
+        }
+
+        String redirectUri = loginProperties.getRedirectUri();
+        if (redirectUri == null) {
+            return DEFAULT_REDIRECT_URI;
+        }
+
+        try {
+            String baseUrl = getBaseUri(redirectUri);
+            return baseUrl + FRONTEND_REDIRECT_URI_PATH;
+        } catch (URISyntaxException e) {
```

This is a rather big error case (misconfiguration of the application)

I think we should not swallow this exception and handle with a default value

Use a error log for such cases and throw the exception

We should know about it

---

Btw: If you choose to define the redirect uri in the properties you should probably move the URI validation there as well